pkg/hubble/metrics/cell/server.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package metricscell
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"sync/atomic"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/hubble/metrics"
+	"github.com/cilium/cilium/pkg/hubble/server/serveroption"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+var (
+	statusWarnWaitCertsAvail = "Waiting for TLS certificates to become available"
+	statusErrWaitCertsAvail  = "Failed waiting for TLS certificates to become available"
+	statusErrServe           = "Failed serving HTTP metrics"
+)
+
+type Server interface {
+	Status() *models.HubbleMetricsStatus
+}
+
+type metricsServer struct {
+	logger *slog.Logger
+
+	server           *http.Server
+	tlsConfigPromise tlsConfigPromise
+
+	statusWarn atomic.Pointer[string]
+	statusErr  atomic.Pointer[string]
+}
+
+func newMetricsServer(p params) Server {
+	if p.Config.MetricsServer == "" {
+		p.Logger.Info("The Hubble metrics server is disabled")
+		return nil
+	}
+
+	mux := metrics.ServerHandler(metrics.Registry, p.Config.EnableOpenMetrics)
+	server := &metricsServer{
+		logger: p.Logger,
+		server: &http.Server{
+			Addr:    p.Config.MetricsServer,
+			Handler: mux,
+		},
+		tlsConfigPromise: p.TLSConfigPromise,
+	}
+
+	p.JobGroup.Add(job.OneShot("hubble-metrics-server", func(ctx context.Context, _ cell.Health) error {
```

Comment:
Bringing to your attention this change where instead of manually spawning a goroutine from lifecycle hooks, I used a one-shot job. Bonus points, we get an entry in the health table for it 🥳